untrusted/quote_verify/src/lib.rs

```diff
@@ -103,6 +117,28 @@ impl Quote {
         }
     }
 
+    pub fn verify_certificate_chain(&self) -> Result<(), Error> {
+        let (_, pem) = pem::parse_x509_pem(ROOT_CERT_PEM)?;
+        let root_cert = pem.parse_x509()?;
+
+        let mut pems =
+            Pem::iter_from_buffer(&self.bytes[QUOTING_ENCLAVE_CERTIFICATION_DATA_START..])
+                .collect::<Result<Vec<_>, _>>()?;
+
+        // Certs are in order from leaf -> .. -> root.  We need to start
+        // verifying from the root.
+        pems.reverse();
+
+        let mut key_bytes = root_cert.public_key().raw.to_vec();
+        for pem in pems.iter() {
+            let cert = pem.parse_x509()?;
+            let (_, key) = SubjectPublicKeyInfo::from_der(key_bytes.as_slice())?;
+            cert.verify_signature(Some(&key))?;
+            key_bytes = cert.public_key().raw.to_vec();
+        }
```

Comment:
Not liking this logic here.  I was fighting getting to the certificates and persisting them around correctly.  Since the logic is:
```mermaid
graph LR
    A(buffer) --> B(PEM) --> C(X509)
```
The lifetime of the buffer doesn't carry through to the X 509 cert structure.